app/models/log.rb

```diff
@@ -7,6 +7,7 @@ class Log < ApplicationRecord
   belongs_to :bulk_upload, optional: true
 
   before_save :update_status!
+  after_find :verify_status!
```

Comment:
We should not mutate objects on find. What are you trying to do here?

Reply:
If we would like to ensure that all existing logs have an up to date status, we can touch them all and `update_status!` will run. I'd be curious to inspect whether this would actually change the status on logs, though?

```ruby
class Log
  def calculate_status
    if all_fields_completed? && errors.empty?
      "completed"
    elsif all_fields_nil?
      "not_started"
    else
      "in_progress"
    end
  end
end

LettingsLog.all.select { |log| log.status != log.calculate_status }.count
```

I ran this on production for you:

```ruby
=> 34
```